lisa/target.py

```diff
@@ -350,17 +350,34 @@ def from_one_conf(cls, path):
         return cls.from_conf(conf=conf, plat_info=plat_info)
 
     @classmethod
-    def from_cli(cls, argv=None) -> 'Target':
+    def from_cli(cls, *args, **kwargs) -> 'Target':
+        """
+        Same as :meth:`from_custom_cli` without the custom parameters
+        capabilities.
+
+        :return: A connected :class:`Target`
+        """
+        args, target = cls.from_custom_cli(*args, **kwargs)
+        return target
+
+    @classmethod
+    def from_custom_cli(cls, argv=None, params=None) -> 'Target':
         """
         Create a Target from command line arguments.
 
         :param argv: The list of arguments. ``sys.argv[1:]`` will be used if
           this is ``None``.
         :type argv: list(str)
 
-        Trying to use this in a script that expects extra arguments is bound
-        to be confusing (help message woes, argument clashes...), so for now
-        this should only be used in scripts that only expect Target args.
+        :param params: Dictionary of custom parameters to add to the parser. It
+            is in the form of
+            ``{param_name: {dict of ArgumentParser.add_argument() options}}``.
+        :type params: dict(str, dict)
+
+        :return: A tuple ``(args, target)``
+
+        .. note:: This method should not be relied upon to implemnt long-term
+            scripts, it's more designed for quick scripting.
```

Comment:
also, with this change, is it really still the case?

Reply:
Before removing that comment, I'd provide an API that works as:
```
parser = LisaArgumentParser(...)
# Target would implement `get_argparse_arguments()` and would build instances with `from_argparse_args()` or something similar
parser.register_class(Target)
...
args = parser.parse_args()
...
target = args.get_instance(Target)
```

This way, the user has full control of the parser, can set the description, the formatter etc. We can also add more "hooks" like this one for `FtraceCollector` for example. They would cooperate for the `--conf` parameter and otherwise add their stuff to the parser.

The issue with that API is that it takes 3 lines for what we currently can do in 1, but that is a much saner way to build a real script out of LISA lego blocks